### PR TITLE
Include project status and related work in readme

### PR DESCRIPTION
**Pull request by lognaturel**
Addresses discussion in https://github.com/XLSForm/pyxform/pull/699 and https://github.com/XLSForm/pyxform/pull/698

#### Why is this the best possible solution? Were any other approaches considered?
It makes minimal updates but hopefully helps clarify who is doing what here and why.

#### What are the regression risks?
None, docs only.

#### Does this change require updates to documentation? If so, please file an issue [here](https://github.com/XLSForm/xlsform.github.io) and include the link below.
No,

#### Before submitting this PR, please make sure you have:
- [x] included test cases for core behavior and edge cases in `tests`
- [x] run `python -m unittest` and verified all tests pass
- [x] run `ruff format pyxform tests` and `ruff check pyxform tests` to lint code
- [x] verified that any code or assets from external sources are properly credited in comments

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -152,3 +167,9 @@ Releases are now automatic. These instructions are provided for forks or for a f
      flit --debug publish --no-use-vcs
 
 6. Tag the GitHub release and publish it.
+
+Related projects
+================
+
+* `xls2xform <https://github.com/mvpdev/xls2xform/>`_: ``pyxform``'s precursor which was rewritten into the library we now know
```
Comment (lognaturel) [inline README.rst:174]: @spwoodcock we could add your Dockerfile repo here

File: README.rst [hunk 2]
```diff
@@ -13,16 +13,27 @@ pyxform
...
+Project status
+===============
+``pyxform`` was started at the `Sustainable Engineering Lab at Columbia University <https://qsel.columbia.edu/>`_ and was then maintained for many years primarily by `Ona <https://ona.io/>`_. Since 2018, it has been maintained by the `ODK team <https://getodk.org/about/team.html>`_ (primarily by `Lindsay Stevens <https://github.com/lindsay-stevens>`_).
 
+Our current goals are:
 
-pyxform is a major rewrite of `xls2xform <http://github.com/mvpdev/xls2xform/>`_.
+* Enabling more complex workflows through sophisticated XPath expressions and `entities <https://getodk.github.io/xforms-spec/entities>`_
+* Improving error messages and making troubleshooting easier
+* Improving experience for multi-language forms
```
Comment (lognaturel) [inline README.rst:26]: @lindsay-stevens Hopefully these capture what we've talked about recently! Maybe you have some to add.

Comment (lindsay-stevens) [reply README.rst]: I think that's about right. If anyone is super keen to know exactly what we're doing there's always the issues list or PR/commit history

**Comment by lindsay-stevens** [comment]
Reworded some of the new content to be a little more assertive, and added a bunch of related projects that I've starred over the years

**Review by lindsay-stevens — APPROVED** [review]

**Comment by lognaturel** [comment]
Thank you!
